storage/drivers/driver.go

```diff
@@ -299,6 +299,15 @@ type DriverWithDiffer interface {
 	DifferTarget(id string) (string, error)
 }
 
+// ApplyDiffStaging is an interface for driver who can apply the diff without holding the main storage lock.
+// This API is experimental and can be changed without bumping the major version number.
+type ApplyDiffStaging interface {
+	// StartStagingDiffToApply applies the layer in a temporary directory. This can be done without holding the storage lock.
```

Comment:
Eventually please copy & paste the detailed description here, e.g. about the unusual `CleanupTempDirFunc` return value.